sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +784,50 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(
+        self,
+        proxy: Literal["files", "ldap"],
+        provider: str | list[str] = "id",
+        proxy_file: str = """""",
+        sever_hostname: str = """""",
+        domain: str | None = None,
+    ):
+        """
+        :param proxy: ``ldap`` or ``files``
+        :type proxy: str
+        :param provider: SSSD providers (id, auth, chpass, ...)
+        :type provider: str | list[str]
+        :param proxy_file: Config for proxy
+        :type proxy_file: str
+        :param sever_hostname: LDAP (LDAP server hostname)
+        :type sever_hostname: str
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+
+        match proxy:
+            case "ldap":
+                target = "sssdproxyldap"
+                self.sssd.fs.write(f"/etc/pam.d/{target}", proxy_file)
+                self.sssd.fs.write(
+                    "/etc/nslcd.conf", f"uid nslcd\ngid ldap\nuri " f"ldap://{sever_hostname}\n", dedent=False
+                )
+                self.sssd.svc.restart("nslcd")
+            case "files":
+                target = "sssdproxyfiles"
+                self.sssd.fs.write(f"/etc/pam.d/{target}", proxy_file)
+            case _:
+                raise Exception("Proxy can be ldap or files but nothing was provided.")
```

Comment:
Can we make files the default behavior? We have a lot more tests that will be using just files instead of nslcd?